uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -465,42 +545,10 @@ define([
      * @param {String} dayType - set to from if from date is selected else to
      * @return {Promise}
      */
-    function loadAbsencePeriodDatesTypes (date, dayType) {
-      var oldPeriodId = vm.period.id;
-      dayType = dayType || 'from';
-      vm.loading[dayType + 'DayTypes'] = true;
-
-      return vm._checkAndSetAbsencePeriod(date)
+    function loadAbsencePeriodDateTypes (date, dateType) {
+      return filterLeaveRequestDayTypes(date, dateType)
         .then(function () {
-          var isInCurrentPeriod = oldPeriodId === vm.period.id;
-
-          if (!isInCurrentPeriod) {
-            // partial reset is required when user has selected a to date and
-            // then changes absence period from from date
-            // no reset required for single days and to date changes
-            if (vm.uiOptions.multipleDays && dayType === 'from') {
-              vm.uiOptions.showBalance = false;
-              vm.uiOptions.toDate = null;
-              vm.request.to_date = null;
-              vm.request.to_date_type = null;
-            }
-
-            return $q.all([
-              vm._loadCalendar()
-            ]);
-          }
-        })
-        .then(function () {
-          vm._setMinMaxDate();
-
-          return filterLeaveRequestDayTypes(date, dayType);
-        })
-        .finally(function () {
-          /**
-           * after the request is completed fromDayTypes or toDayTypes are
-           * set to false and the corresponding field is shown on the ui.
-           */
-          vm.loading[dayType + 'DayTypes'] = false;
+          vm.loading[dateType + 'DayTypes'] = false;
```

Comment:
where is the
```js
vm.loading[dateType + 'DayTypes'] = true;
```
? Wouldn't it make sense to have it at the top of the function?

Reply:
Not really. Here is the flow:
1) date changed
2) We need to show in UI that the date type is loading
3) If absence period has changed then we need to reload entitlements, this happens in master controller and we **do not have a control over it**.
4) then once entitlements are updated - we load day types and just then telling UI the loading is finished.

The `=true` is in resetAfterDateChange() function.